### DISCUSSION TITLE
How to use callbacks with DistributedDataParallel training?

### DISCUSSION THREAD

**Discussion by haimat** [ultralytics/yolov5#8325] [General]
We use the callback system in the YOLOv5 training function `train()` from `train.py` file in order to integrate it into our custom training frontend. This works fine, even when using **Multi-GPU DataParallel Mode**, running in one process. Now we would like to train in **Multi-GPU DistributedDataParallel Mode**, however, as far as I understand this would spawn multiple processes, right? So if this is the case, how can one train in DistributedDataParallel mode but still make use of the callbacks in the training script in order to get current status updates during training?

**Comment by glenn-jocher** [comment]
@haimat we don't assist with code customizations but you might visit Multi-GPU Training tutorial for DDP examples commands.


## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️  RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Roboflow for Datasets, Labeling, and Active Learning](https://github.com/ultralytics/yolov5/issues/4975)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [TFLite, ONNX, CoreML, TensorRT Export](https://github.com/ultralytics/yolov5/issues/251) 🚀
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [Architecture Summary](https://github.com/ultralytics/yolov5/issues/6998)&nbsp; ⭐ NEW

Good luck 🍀 and let us know if you have any other questions!
